### ISSUE TITLE
0 - Class 'Jstring' not found after updating to php 7 & Joomla 3.7.0 & 3.7.1

### ISSUE THREAD

**Issue by ObsessedGamer** [joomla/joomla-cms#16096]
Certain pages give this error now, home page & contact-us page works fine but other links give this error.  After downgrading to 3.5x it loads right up all pages.  Thanks for any insight!

Edit: Also I downgraded PHP to previous 5.3 versions and the same articles giving that error on PHP 7 are just now blank.

**Comment by mbabker** [comment]
The Joomla core autoloader requires that developers use case sensitive names matching our class declarations (and inherently the filesystem structure based on its rules) if a class hasn't been loaded into memory.  We don't have a `Jstring` class in core, but there is `JString`, and that one will correctly autoload.  So scan your code and see if there are any class declarations that don't use the correct casing and fix those (or report the issue to the extension developer in question).

**Comment by ObsessedGamer** [comment]
I am sorry but I am not very literate when it comes to the coding site of the files in my installation.  Can you specify where I would look for this?

**Comment by Bakual** [comment]
> Just to be sure you are saying this is actually a extension causing this and not anything to do with the Joomla core files?

The exact error message would help here. But changes are very high that it is an extension (eg a content or system plugin) causing this.

I'm closing this issue. If you can reproduce the issue on a clean Joomla installation it can be reopened,

**Comment by ObsessedGamer** [comment]
I have disabled all extensions but the core joomla ones and the error still persists.  The exact error is 0 - Class 'Jstring' not found.

**Comment by Bakual** [comment]
We would need exakt steps to reproduce this on a clean Joomla install.
Disabling extensions isn't th same thing.

**Comment by franz-wohlkoenig** [comment]
reopened as Discussion going on.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16096">issues.joomla.org/tracker/joomla-cms/16096</a>.</sub>

**Comment by ObsessedGamer** [comment]
I am doing a brand new clean installation now and will report back ASAP.  Thanks for re-opening!

**Comment by ObsessedGamer** [comment]
I have narrowed it down to 1 specific module that does not get along with Joomla 3.7 or 3.7.1 - JoomlaXTC Deluxe News Pro , I disabled that module & all pages load right up.  some of the errors I am getting are as follows: 

A parse error has occurred while processing the following JSON data:
<!DOCTYPE html>
<html lang="en-gb" dir="ltr">
<head>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta http-equiv="X-UA-Compatible" content="IE=edge"/>
<meta charset="utf-8"/>
<meta name="generator" content="Joomla! - Open Source Content Management"/>
//<![CDATA[
try{if (!window.CloudFlare) {var CloudFlare=[{verbose:0,p:1495216073,byc:0,owlid:"cf",bag2:1,mirage2:0,oracle:0,paths:{cloudflare:"/cdn-cgi/nexp/dok3v=1613a3a185/"},atok:"1d4c5a420db9be7f672d05a973e4d80c",petok:"d34178a609968a0e5b45134dbc6850d9eb257512-1495216446-1800",zone:"savemoneyonenergy.com",rocket:"0",apps:{"ga_key":{"ua":"UA-2300463-1","ga_bs":"2"}}}];!function(a,b){a=document.createElement("script"),b=document.getElementsByTagName("script")[0],a.async=!0,a.src="//ajax.cloudflare.com/cdn-cgi/nexp/dok3v=85b614c0f6/cloudflare.min.js",b.parentNode.insertBefore(a,b)}()}}catch(e){};
//]]>
</script>
<link href="/administrator/templates/isis/favicon.ico" rel="shortcut icon" type="image/vnd.microsoft.icon"/>
<link href="/media/regularlabs/css/style.min.css?v=17.5.1388" rel="stylesheet"/>
<link href="/media/betterpreview/css/style.min.css?v=5.0.1" rel="stylesheet"/>
<link href="/administrator/templates/isis/css/template.css?2e60e68fb95354eb9c457ede7e3eb866" rel="stylesheet"/>
<style>#scrollToTop{cursor:pointer;font-size:0.9em;position:fixed;text-align:center;z-index:9999;-webkit-transition:background-color 0.2s ease-in-out;-moz-transition:background-color 0.2s ease-in-out;-ms-transition:background-color 0.2s ease-in-out;-o-transition:background-color 0.2s ease-in-out;transition:background-color 0.2s ease-in-out;background:#1500ff;color:#518f3e;border-radius:3px;padding-left:12px;padding-right:12px;padding-top:12px;padding-bottom:12px;right:20px;bottom:20px;}#scrollToTop:hover{background:#ffffff;color:#ffffff;}#scrollToTop>img{display:block;margin:0 auto;}#gototop{border:1px solid #000;margin:5px;padding:5px;background-color:#ff0;color:#000;}#gototop:hover{background-color:#000;color:#ff0;}@media (max-width: 480px) {.view-login .container{margin-top:-170px;}.btn{font-size:13px;padding:4px 10px 4px;}}</style>
<script type="application/json" class="joomla-script-options new">{"system.keepalive":{"interval":3600000,"uri":"\/administrator\/index.php"}}</script>
<script src="/media/jui/js/jquery.min.js?2e60e68fb95354eb9c457ede7e3eb866"></script>
<script src="/media/jui/js/jquery-noconflict.js?2e60e68fb95354eb9c457ede7e3eb866"></script>
<script src="/media/jui/js/jquery-migrate.min.js?2e60e68fb95354eb9c457ede7e3eb866"></script>
<script src="/media/jui/js/bootstrap.min.js?2e60e68fb95354eb9c457ede7e3eb866"></script>
<script src="/media/regularlabs/js/script.min.js?v=17.5.1388"></script>
<script src="/media/betterpreview/js/script.min.js?v=5.0.1"></script>
<script src="/media/plg_system_sl_scrolltotop/js/scrolltotop_jq.js"></script>
<script src="/media/system/js/core.js?2e60e68fb95354eb9c457ede7e3eb866"></script>

<script src="/media/system/js/keepalive.js?2e60e68fb95354eb9c457ede7e3eb866"></script>

<script>
jQuery(function($){ $(".hasTooltip").tooltip({"html": true,"container": "body"}); });
jQuery(document).ready(function() {

	jQuery(document.body).SLScrollToTop({

		'image':		'',

		'text':			'Back to top',

		'title':		'Back to top',

		'className':	'scrollToTop',

		'duration':		500

	});

});
	</script>
<script type="text/javascript">
/* <![CDATA[ */
var _gaq = _gaq || [];
_gaq.push(['_setAccount', 'UA-2300463-1']);
_gaq.push(['_trackPageview']);

(function() {
var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
})();

(function(b){(function(a){"__CF"in b&&"DJS"in b.__CF?b.__CF.DJS.push(a):"addEventListener"in b?b.addEventListener("load",a,!1):b.attachEvent("onload",a)})(function(){"FB"in b&&"Event"in FB&&"subscribe"in FB.Event&&(FB.Event.subscribe("edge.create",function(a){_gaq.push(["_trackSocial","facebook","like",a])}),FB.Event.subscribe("edge.remove",function(a){_gaq.push(["_trackSocial","facebook","unlike",a])}),FB.Event.subscribe("message.send",function(a){_gaq.push(["_trackSocial","facebook","send",a])}));"twttr"in b&&"events"in twttr&&"bind"in twttr.events&&twttr.events.bind("tweet",function(a){if(a){var b;if(a.target&&a.target.nodeName=="IFRAME")a:{if(a=a.target.src){a=a.split("#")[0].match(/[^?=&]+=([^&]*)?/g);b=0;for(var c;c=a[b];++b)if(c.indexOf("url")===0){b=unescape(c.split("=")[1]);break a}}b=void 0}_gaq.push(["_trackSocial","twitter","tweet",b])}})})})(window);
/* ]]> */
</script>
</head>
<body class="site com_login view-login layout-default task- itemid- ">
 
<div class="container">
<div id="content">

Is there any way to make this module compatible with the new Joomla?

**Comment by wojsmol** [comment]
@ObsessedGamer Please report the issue to JoomlaXTC support

**Comment by franz-wohlkoenig** [comment]
closed as no Core Issue.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16096">issues.joomla.org/tracker/joomla-cms/16096</a>.</sub>

**Comment by brianteeman** [comment]
This may help https://github.com/renekreijveld/JoomlaJsonCheck

**Comment by ObsessedGamer** [comment]
Thank you for the help guys & Brian I am going to checkout that link, thanks!
